contracts/Strategy.sol

```diff
@@ -603,7 +598,7 @@ contract Strategy is BaseStrategy {
 
         uint256 price = uint256(chainlinkWantUnderlyingTokenToETHPriceFeed.latestAnswer());
         //TODO: check if this formula is working properly
-        return _amtInWei.mul(EXCHANGE_RATE_PRECISION).div(price).mul(wantAsVault.pricePerShare());
+        return _amtInWei.mul(EXCHANGE_RATE_PRECISION).mul(wantAsVault.pricePerShare()).div(price);
```

Comment:
`ethToWant` is probably not that important, but I think decimals are incorrect here. Code in `master` does only work for 18 decimals tokens (no USDC for example). Moreover, here you are adding a multiplication with `pricePerShare` which will include additional decimals.

Reply:
Moreover, what is the purpose of `EXCHANGE_RATE_PRECISION`? Does abra always return 18 decimals?

Reply:
Yes, I still need to fix this. But I am using the exchange rate precision because we are using abra as a the price oracle, too.